Models/Key.js

```diff
@@ -0,0 +1,141 @@
+import { AsyncStorage } from 'react-native'
+
+const keyIds = 'keyIds'
+
+export default class KeyModel {
+
+    /**
+     * Will create or update an existing key with new data.
+     * @param {object} KeyData 
+     */
+    static async addOrUpdateKey(keyData) {
+        return KeyModel.getAllKeyIds()
+            .then((Ids) => {
+                let idArray = []
+                if (Ids) {
+                    // Need to check if there's only one Id
+                    if (Ids.includes(keyData.id) || (keyData.id === Ids)) {
+                        return
+                    }
+                    idArray = Ids
+                } else {
+                    idArray = []
+                }
+                idArray.push(keyData.id)
+                const idString = JSON.stringify(idArray)
+                return AsyncStorage.setItem(keyIds, idString)
+            })
+            .then(() => {
+                const dataString = JSON.stringify(keyData)
+                return AsyncStorage.setItem(keyData.id, dataString)
+            })
+    }
+
+    /**
+     * Get key data for a given keyId.
+     * @param {string} keyId
+     */
+    static async getKeyWithId(keyId) {
+        return AsyncStorage.getItem(keyId)
+            .then((result) => {
+                return result
+            })
+    }
+
+    /**
+     * Get all the Ids for keys that have been created.
+     */
+    static async getAllKeyIds() {
+        return AsyncStorage.getItem(keyIds)
+            .then((result) => {
+                if (!result) {
+                    // Back out if we don't have any Ids.
+                    return result
+                }
+                const resultArray = JSON.parse(result)
+                return resultArray
+            })
+    }
+
+    /**
+     * Get all the key data for the Ids stored.
+     */
+    static async getAllKeyData() {
+        // Fetch the array of all the key Ids.
+        return KeyModel.getAllKeyIds()
+            .then((result) => {
+                if (!result) {
+                    // Back out if we don't have any Ids.
+                    return result
+                }
+ 
+                resultArray = JSON.parse(result)
```

Comment:
The result from getAllKeyIds is already parsed, so we shouldn't need to do that again